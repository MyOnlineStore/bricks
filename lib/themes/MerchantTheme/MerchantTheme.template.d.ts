import { SFC } from 'react';
import ThemeType from '../../types/ThemeType';
declare type PropsType = {
    theme: ThemeType;
};
declare const MerchantTheme: SFC<PropsType>;
export default MerchantTheme;
export { PropsType };
