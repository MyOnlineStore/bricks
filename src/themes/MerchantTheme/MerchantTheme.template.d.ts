import { FunctionComponent } from 'react';
import ThemeType from '../../types/ThemeType';
declare type PropsType = {
    theme: ThemeType;
};
declare const MerchantTheme: FunctionComponent<PropsType>;
export default MerchantTheme;
export { PropsType };
