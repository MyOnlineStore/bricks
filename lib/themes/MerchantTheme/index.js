!function(e,t){"object"==typeof exports&&"object"==typeof module?module.exports=t(require("react"),require("styled-components")):"function"==typeof define&&define.amd?define("bricks",["react","styled-components"],t):"object"==typeof exports?exports.bricks=t(require("react"),require("styled-components")):e.bricks=t(e.react,e["styled-components"])}("undefined"!=typeof self?self:this,function(e,t){return function(e){var t={};function r(n){if(t[n])return t[n].exports;var o=t[n]={i:n,l:!1,exports:{}};return e[n].call(o.exports,o,o.exports,r),o.l=!0,o.exports}return r.m=e,r.c=t,r.d=function(e,t,n){r.o(e,t)||Object.defineProperty(e,t,{configurable:!1,enumerable:!0,get:n})},r.n=function(e){var t=e&&e.__esModule?function(){return e.default}:function(){return e};return r.d(t,"a",t),t},r.o=function(e,t){return Object.prototype.hasOwnProperty.call(e,t)},r.p="",r(r.s=638)}({0:function(t,r){t.exports=e},1:function(e,t,r){"use strict";Object.defineProperty(t,"__esModule",{value:!0}),r.d(t,"css",function(){return c}),r.d(t,"injectGlobal",function(){return i}),r.d(t,"keyframes",function(){return f}),r.d(t,"ThemeProvider",function(){return s}),r.d(t,"withProps",function(){return d});var n=r(2),o=(r.n(n),n),u=o.default,c=o.css,i=o.injectGlobal,f=o.keyframes,s=o.ThemeProvider;function d(e){return e}t.default=u},2:function(e,r){e.exports=t},638:function(e,t,r){"use strict";Object.defineProperty(t,"__esModule",{value:!0});var n=r(639);r.d(t,"default",function(){return n.a})},639:function(e,t,r){"use strict";var n=r(0),o=r.n(n),u=r(1);t.a=function(e){return o.a.createElement(u.ThemeProvider,{theme:e.theme},e.children)}}})});
//# sourceMappingURL=index.js.map