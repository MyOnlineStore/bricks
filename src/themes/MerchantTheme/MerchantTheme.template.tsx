import React, { FunctionComponent } from 'react';
import ThemeType from '../../types/ThemeType';
import { ThemeProvider } from '../../utility/_styled';

type PropsType = {
    theme: ThemeType;
};

const MerchantTheme: FunctionComponent<PropsType> = (props): JSX.Element => (
    <ThemeProvider theme={props.theme}>{props.children}</ThemeProvider>
);

export default MerchantTheme;
export { PropsType };
