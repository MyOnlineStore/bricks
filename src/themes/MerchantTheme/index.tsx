import React, { FunctionComponent } from 'react';
import ThemeType from '../../types/ThemeType';
import { ThemeProvider } from '../../utility/_styled';
import mosTheme from '../MosTheme/MosTheme.theme';
import deepmerge from 'deepmerge';
import RecursivePartialType from '../../types/_RecursivePartialType';

type PropsType = {
    theme: RecursivePartialType<ThemeType>;
};

const customTheme = (theme: RecursivePartialType<ThemeType>): ThemeType => {
    return deepmerge(mosTheme, theme) as ThemeType;
};

const MerchantTheme: FunctionComponent<PropsType> = (props): JSX.Element => (
    <ThemeProvider theme={customTheme(props.theme)}>
        <>{props.children}</>
    </ThemeProvider>
);

export default MerchantTheme;
export { PropsType };
