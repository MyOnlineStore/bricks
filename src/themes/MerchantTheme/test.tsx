import { mount, shallow } from 'enzyme';
import React from 'react';
import MerchantTheme from '../MerchantTheme';
import Badge from '../../components/Badge';
import Heading from '../../components/Heading';
import ThemeType from '../../types/ThemeType';
import { ThemeProvider } from '../../utility/_styled/testing';
import 'jest-styled-components';
import RecursivePartialType from '../../types/_RecursivePartialType';

describe('MerchantTheme', () => {
    const heading = {
        color: 'mockValue',
        fontFamily: 'mockValue',
        fontSize: 'mockValue',
        fontWeight: 'mockValue',
        lineHeight: 'mockValue',
        textTransform: 'mockValue',
    };

    const theme: RecursivePartialType<ThemeType> = {
        Heading: [heading, heading, heading, heading, heading, heading],
    };

    // tslint:disable-next-line
    const context = (shallow(<MerchantTheme theme={theme} />) as any)
        .dive()
        .instance()
        .getChildContext();

    it('should render a component with merchant theme', () => {
        const fn = () => {
            return mount(<Heading />, {
                context,
                childContextTypes: ThemeProvider.childContextTypes,
            });
        };

        expect(fn).not.toThrow();
        expect(fn()).toHaveStyleRule('color', 'mockValue');
    });

    it('should fallback on MosTheme when theme properties are undefined', () => {
        expect(() => {
            mount(<Badge />, {
                context,
                childContextTypes: ThemeProvider.childContextTypes,
            });
        }).not.toThrow();
    });
});
