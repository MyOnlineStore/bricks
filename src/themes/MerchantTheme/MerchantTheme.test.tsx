import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import MerchantTheme from '../MerchantTheme';
import { PropsType } from './MerchantTheme.template';

describe('MerchantTheme', () => {
    it('should render the merchant theme', () => {
        const theme = {
            Heading: {
                default: {
                    color: 'mockValue',
                    fontFamily: 'mockValue',
                    fontSize: 'mockValue',
                    fontWeight: 'mockValue',
                    lineHeight: 'mockValue',
                    textTransform: 'mockValue',
                },
                hierarchy: [
                    {
                        color: 'mockValue',
                        fontFamily: 'mockValue',
                        fontSize: 'mockValue',
                        fontWeight: 'mockValue',
                        lineHeight: 'mockValue',
                        textTransform: 'mockValue',
                    },
                    {
                        color: 'mockValue',
                        fontFamily: 'mockValue',
                        fontSize: 'mockValue',
                        fontWeight: 'mockValue',
                        lineHeight: 'mockValue',
                        textTransform: 'mockValue',
                    },
                    {
                        color: 'mockValue',
                        fontFamily: 'mockValue',
                        fontSize: 'mockValue',
                        fontWeight: 'mockValue',
                        lineHeight: 'mockValue',
                        textTransform: 'mockValue',
                    },
                    {
                        color: 'mockValue',
                        fontFamily: 'mockValue',
                        fontSize: 'mockValue',
                        fontWeight: 'mockValue',
                        lineHeight: 'mockValue',
                        textTransform: 'mockValue',
                    },
                    {
                        color: 'mockValue',
                        fontFamily: 'mockValue',
                        fontSize: 'mockValue',
                        fontWeight: 'mockValue',
                        lineHeight: 'mockValue',
                        textTransform: 'mockValue',
                    },
                    {
                        color: 'mockValue',
                        fontFamily: 'mockValue',
                        fontSize: 'mockValue',
                        fontWeight: 'mockValue',
                        lineHeight: 'mockValue',
                        textTransform: 'mockValue',
                    },
                ],
            },
        };

        const component = shallow(<MerchantTheme theme={theme as PropsType['theme']} />);

        expect(toJson(component)).toMatchSnapshot();
    });
});
