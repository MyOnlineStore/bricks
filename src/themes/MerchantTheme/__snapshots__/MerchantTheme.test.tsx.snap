// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MerchantTheme should render the merchant theme 1`] = `
<ThemeProvider
  theme={
    Object {
      "Heading": Object {
        "default": Object {
    +           "color": "mockValue",
    +           "fontFamily": "mockValue",
    +           "fontSize": "mockValue",
    +           "fontWeight": "mockValue",
    +         },
    +         "subHeading": Object {
    +           "color": "mockValue",
    +           "fontFamily": "mockValue",
    +           "fontSize": "mockValue",
    +           "fontWeight": "mockValue",
    +         },
      },
    }
  }
/>
`;
